lib/graphql/execution/interpreter/runtime.rb

```diff
@@ -675,6 +675,8 @@ def continue_field(value, owner_type, field, current_type, ast_node, next_select
           when "SCALAR", "ENUM"
             r = begin
               current_type.coerce_result(value, context)
+            rescue GraphQL::ExecutionError => ex_err
+              return continue_value(ex_err, field, is_non_null, ast_node, result_name, selection_result)
```

Comment:
Any possible issues here? No tests break but this will change error messages, though it will improve and make them more spec compliant.